src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+import core.basesyntax.shape.Shape;
+import core.basesyntax.supplier.ShapeSupplier;
+
+public class Main {
+    public static void main(String[] args) {
+        ShapeSupplier shapeSupplier = new ShapeSupplier();
+
+        Shape[] shapes = new Shape[] {
```

Comment:
It is recommended to use List (e.g. ArrayList) instead of an array to store the shapes for better flexibility and usage of collection methods.